trin-core/src/portalnet/types.rs

```diff
@@ -345,11 +345,405 @@ impl FromStr for HexData {
+                )),
+                0x04 => Ok(StateContentKey::ContractBytecodeKey(
+                    ContractBytecode::from_ssz_bytes(&bytes[1..])
+                      .map_err(|e| format!("Failed to decode ContractBytecode ssz: {:?}. The bytes attempted to decode are: {:?}", e, bytes))?
+                )),
+                _ => Err(format!("No matching content-type for {} for StateContentKey", content_type)),
+            }
+        } else {
+            Err("Cannot decode StateContentKey from empty bytes".to_string())
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum HistoryContentKey {
+    HeaderKey(HeaderKey),
+    BodyKey(BodyKey),
+    ReceiptsKey(ReceiptsKey),
+    // TODO: add transaction key
+}
+
+impl HistoryContentKey {
+    pub fn content_type(&self) -> u8 {
+        match self {
+            HistoryContentKey::HeaderKey(_) => 0x01,
+            HistoryContentKey::BodyKey(_) => 0x02,
+            HistoryContentKey::ReceiptsKey(_) => 0x03,
+        }
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, String> {
+        if let Some(content_type) = bytes.first() {
+            match content_type {
+              0x01 => Ok(HistoryContentKey::HeaderKey(
+                  HeaderKey::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode HeaderKey ssz: {:?}. The bytes attempted to decode are: {:?}", e, bytes))?
+              )),
+              0x02 => Ok(HistoryContentKey::BodyKey(
+                  BodyKey::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode BodyKey ssz: {:?}. The bytes attempted to decode are: {:?}", e, bytes))?
+              )),
+              0x03 => Ok(HistoryContentKey::ReceiptsKey(
+                  ReceiptsKey::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode ReceiptsKey ssz: {:?}. The bytes attempted to decode are: {:?}", e, bytes))?
+              )),
+              _ => Err(format!("No matching content-type for {} for HistoryContentKey", content_type)),
+            }
+        } else {
+            Err("Cannot decode HistoryContentKey from empty bytes".to_string())
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Encode, Decode)]
+pub struct HeaderKey {
+    pub chain_id: u16,
+    pub block_hash: [u8; 32],
+}
+
+#[derive(Debug, Clone, PartialEq, Encode, Decode)]
+pub struct BodyKey {
+    pub chain_id: u16,
+    pub block_hash: [u8; 32],
+}
+
+#[derive(Debug, Clone, PartialEq, Encode, Decode)]
+pub struct ReceiptsKey {
+    pub chain_id: u16,
+    pub block_hash: [u8; 32],
+}
+
+#[derive(PartialEq, Clone, Debug, Encode, Decode)]
+pub struct AccountTrieNode {
+    pub path: Vec<u8>,
+    pub node_hash: U256,
+    pub state_root: [u8; 32],
+}
+
+#[derive(Debug, Clone, PartialEq, Encode, Decode)]
+pub struct ContractStorageTrieNode {
+    pub address: Vec<u8>, // should be bytes20
```

Comment:
I think you mean `FixedVector`? Since this needs 20 bytes and 20 bytes only. Also, since we can now start using `ssz_types` this raises the larger question of if we want to replace all types with `ssz_types` implemented in the crate? For example, for a bytes32, we can use a bare rust type of `[u8; 32]` or use `ssz_types` of `FixedVector<u8, typenum::U32`. Having the bare type makes things simpler but using the ssz_type gives us additional flexibility.